chapters/chapter-abstracts.md

```diff
@@ -0,0 +1,147 @@
+Make executes the recipe associated with its rule.
+Make defines a few convenience variables
+such as `$@` (target),
+`$^` (all prerequisites),
+and `$<` (first prerequisite).
+Makefiles can also use wildcard matching of file names
+and functions such as `$(patsubst ...)` for increased flexibility.
+As always it is important to document our code
+and in Make we use specially-formatted comments to create self-documenting Makefiles.
+
+## 10. Configuring Programs
+
+For complex programs
+it can improve usability and increase reproducibility
+to be able to read configuration options from files
+in addition to using command-line options.
+We may want to use up to four layers of configuration:
+a system-wide configuration file for general settings;
+a user-specific configuration file for personal preferences;
+a job-specific file with settings for a particular run;
+and command-line options to change things that commonly change.
+This is sometimes called overlay configuration
+because each level overrides the ones above it.
+Often,
+a pre-existing standard such as the YAML or INI syntax
+is used to write these configuration files.
+Saving and sharing these configuration files
+makes it easy for researchers programs with the same settings,
+which reduces the risk for errors and increases reproducibility.
+
+## 11. Testing Software
+
+It is critical to test software to reassure people (including yourself)
+that the software is correct enough
+and to make tolerances on "enough" explicit.
+There are several layers to testing software:
+adding assertions to code so that it checks itself as it runs;
+writing unit tests to check individual pieces of code;
+writing integration tests to check that those pieces work together correctly;
+writing regression tests to check if things that used to work no longer do.
+Fortunately,
+there are many excellent test frameworks to aid us with these tasks.
+A test framework finds and runs tests written in a prescribed fashion
+and reports their results.
+In addition to test failures,
+these reports often include the tests' coverage,
+which is the fraction of lines of code that are checked.
+To automate the execution of our tests
+we can use Continuous Integration
+which runs tests every time we make changes to the code,
+so that we can immediately spot when something breaks.
+
+## 12. Handling errors
+
+People will misunderstand how to use our programs,
+and we should therefore plan from the start to detect and handle errors.
+Something that goes wrong while a program is running
+is sometimes referred to as an exception from normal behavior.
+Generally speaking,
+we distinguish between two types of errors/exceptions.
+Internal errors are mistakes in the program itself,
+such as calling a function with None instead of a list.
+External errors are usually caused by interactions
+between the program and the outside world:
+a user may mis-type a filename,
+the network might be down,
+and so on.
+
```

Comment:
This handling errors abstract is incomplete. It needs to talk about all the things we do in this chapter, like raising and catching exceptions, logging, etc.